ifrs17/Import/Importers.ipynb

```diff
@@ -673,44 +718,64 @@
-                "\n    ",
-                "\n    var toCommitYieldCurves = await workspace.Query<YieldCurve>().ToArrayAsync();",
-                "\n    if (!toCommitYieldCurves.Any()){",
+                "\n    var toCommitYieldCurves  = await workspace.Query<YieldCurve>().ToArrayAsync();",
+                "\n    if (!toCommitYieldCurves.Any()) {",
                 "\n        ApplicationMessage.Log(Warning.VariablesAlreadyImported);   ",
                 "\n        return Activity.Finish().Merge(importLog);",
                 "\n    }",
-                "\n    ",
-                "\n    await DataSource.UpdateAsync<YieldCurve>(toCommitYieldCurves);",
-                "\n    await DataSource.CommitAsync();",
+                "\n",
+                "\n    var allArgs = await GetAllArgsAsync(primaryArgs, options.TargetDataSource, ImportFormats.YieldCurve);",
+                "\n    var updatedCurrencies = toCommitYieldCurves.Select(x => x.Currency).Distinct();",
+                "\n    var dataNodesToUpdate = await workspace.Query<GroupOfContract>().Where(x => updatedCurrencies.Contains(x.ContractualCurrency)).Select(x => x.SystemName).ToArrayAsync();",
+                "\n    var workspaceToCompute = Workspace.CreateNew();",
+                "\n    workspaceToCompute.Initialize(x => x.FromSource(options.TargetDataSource));",
                 "\n   ",
+                "\n    foreach (var args in allArgs) {",
+                "\n        await CommitPartitionAsync<PartitionByReportingNodeAndPeriod>(args, workspace, workspaceToCompute);",
+                "\n        var targetPartition = (Guid)(await options.TargetDataSource.Partition.GetKeyForInstanceAsync<PartitionByReportingNodeAndPeriod>(args));",
+                "\n        var defaultPartition = (Guid)(await options.TargetDataSource.Partition.GetKeyForInstanceAsync<PartitionByReportingNodeAndPeriod>(",
+                "\n            new ImportArgs(args.ReportingNode, args.Year, args.Month, default(Periodicity), null, ImportFormats.Cashflow) ));",
+                "\n        if(ApplicationMessage.HasErrors()) return Activity.Finish().Merge(importLog);",
+                "\n        ",
+                "\n        // Avoid starting the computation if no best estimate cash flow has ever been imported ",
+                "\n        if(args.Scenario == null) {",
+                "\n            await options.TargetDataSource.Partition.SetAsync<PartitionByReportingNodeAndPeriod>(null);",
+                "\n            if(!(await options.TargetDataSource.Query<RawVariable>().Where(x => x.Partition == targetPartition).Take(1).ToArrayAsync()).Any()) continue;",
+                "\n        }",
+                "\n",
+                "\n        // Remove data nodes which are unaffected by the updated yield curves",
+                "\n        await workspaceToCompute.DeleteAsync( await workspaceToCompute.Query<RawVariable>()",
```

Comment:
what is the type that is deleted here? is it RawVariable? why?

Reply:
The yield curve importer triggers calculation with ImportFormat Cashflow. Therefore, the parsedRawVar and parsedIfrsVar in the ImportStorage will be full and empty respectively. However, the former can be pruned in order to select only those raw variables connected to data nodes having the same currency as one of those newly imported. This is why we have this step here where we delete useless raw variables.